libs/effects-ng/src/lib/use-directive-effects.ts

```diff
@@ -0,0 +1,81 @@
+import { Directive, Type, inject, OnDestroy } from '@angular/core';
+import { EFFECTS_MANAGER } from './tokens';
+import { getEffectPropsMap } from './utils';
+import { Effect } from '@ngneat/effects';
+import {
+  isEffectProvided,
+  increaseProvidedEffectSources,
+  provideEffect,
+  decreaseProvidedEffectSources,
+  generateProvidedEffectToken,
+  ProvidedEffectToken,
+  getProvideEffectByToken,
+} from './provided-effects-map';
+
+export function useDirectiveEffects(...providers: Type<any>[]): Type<unknown> {
+  @Directive({
+    standalone: true,
+    providers,
+  })
+  class InitEffectsDirective implements OnDestroy {
+    private readonly manager = inject(EFFECTS_MANAGER, { optional: true });
+    private readonly providersEffectsTokens = new Set<ProvidedEffectToken>();
+
+    constructor() {
+      if (!this.manager) {
+        throw new TypeError(
+          "Effects manager wasn't found. Have you forgotten to provide it? Please call `provideEffectsManager` in the root providers."
+        );
+      }
+
+      const effects: Effect[] = [];
+
+      new Set(providers).forEach((provider) => {
+        const instance = inject(provider, { self: true });
+
+        getEffectPropsMap(instance).forEach((effect, key) => {
```

Comment:
Can we go DRY with this chunk? It's used in multiple places

Reply:
I can expose this chunk oof cod for `module` implementation until it's removed.